docs/pages/core-concepts.mdx

```diff
@@ -0,0 +1,219 @@
+---
+title: Teleport Core Concepts
+description: Learn the the key architectural components that make up Teleport.
+---
+
+Here are the core concepts that make up a Teleport deployment. You will see
+these terms often when setting up and managing Teleport, so you should get
+familiar with them before following other pages in the documentation.
+
+## Authentication connector
```

Comment:
I think it would improve readability and comprehension if we pulled out the primary terminologies into its own section at the top and then have an alphabetical list of terminologies in a secondary section. For example, we should pull Teleport Cluster, Teleport Process, and Teleport Service out and put them at the top so users understand the fundamental pieces of Teleport before reading further.